### ISSUE TITLE
Prevent multiple people from publishing the same raid

### ISSUE THREAD

**Issue by pbijdens** [pbijdens/raidbot#1]
It would be nice if it would be possible to prevent multiple people from publishing the same raid. 

**Implementation suggestions:**
Since people have to manually enter the name of the Pokémon and  Gym, using those to detect the situation is probably not the most reliable solution. We could use the location of the Gym and then warn a user that the raid is already published before they 'Save' it. Also we can render a warning triangle on the publish button if another raid was already published nearby.

**Comment by FractionalRadix** [comment]
If two raids are in gyms with very similar names, we could warn the user ("A very similar raid is already published. Are you sure this is a different raid?") 
We could use Levenshtein distance for this purpose; possibly removing things like "The"/"De" from a gym name before doing the comparison.

We could also have a file of known gyms, and compare the input to these.

**Comment by pbijdens** [comment]
Agreed. Those are solid ideas. I think we need not even use the name of the
raid in the comparison, the odds that on another gym at roughly the same
location (40 m apart) at almost exactly the same time a different raid
occurs are sufficiently low. I will see what I can do.

As for gym names, I am considering adding a custom keyboard for previously
used gym names at that location.
